lib/graphiti/delegates/pagination.rb

```diff
@@ -53,7 +53,7 @@ def last_page
       def item_count
         begin
           return @item_count if @item_count
-          @item_count = @proxy.resource.stat(:total, :count).call(@proxy.scope.unpaginated_object, :total)
+          @item_count = item_count_from_proxy || item_count_from_stats
```

Comment:
If `stats: { total: 'count' }` is present in params then try to fetch the count using `@proxy.stats` because `@proxy.stats` is memoized so later on it will be reused again and won't fire duplicate count query again.

else if `stats: { total: 'count' }` is not present in params then explicitly calculate count using `item_count_from_stats` method